azerothcore/azerothcore-wotlk
# fix(Core/Spell): solve issue with Life Tap All Ranks

## Changes Proposed:
-  I fixed the fizzle, at 20% it get activates (approx) depends on the level / HP (Life Tap Rank 1)
-  I fixed the amount of life subtracted from the spell to convert to mana, now each rank of "Life Tap" subtracts the same amount of life it adds to mana previously it subtracted 34 life points, now it subtracts 27 points(as it should does with Rank 1) 
and in order also the other Ranks:
Rank 2 | 66
Rank 3 | 132
Rank 4 | 215
Rank 5 | 306
Rank 6 | 827
Rank 7 | 1124
Rank 8 | 2000



[EDIT]
- added SQL query in the "spell_dbc" table of Life Tap (All Ranks)
Fix mana returned and correction Tooltip


## Issues Addressed:
- Closes 



## Tests Performed:
-Compiled without errors
-Tested in game and working
https://youtu.be/5X3usUTZiJI **(New! Video! English Version)** **(Update 05/February/2021 )**
-Tested on Windows 10 (64)



## How to Test the Changes:


- Check the amount of life subtracted from the spell and the fizzle.
- Check the amount of mana returned
- https://youtu.be/5X3usUTZiJI My Test and Investigation about the Spell.

## Known Issues and TODO List:

- [x] To complete this fix I also had to patch the Game Client to make the fix made appear "Visually" https://github.com/azerothcore/azerothcore-wotlk/discussions/4462
This is the post with the suggestion I gave for the Game Client Fixes


## Target Branch(es):








 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
